modules/imgproc/src/morph.cpp

```diff
@@ -1615,6 +1615,7 @@ static bool ocl_morphSmall( InputArray _src, OutputArray _dst, InputArray _kerne
 
     int idxArg = kernel.set(0, ocl::KernelArg::PtrReadOnly(source));
     idxArg = kernel.set(idxArg, (int)source.step);
+    CV_Assert(source.elemSize() > 0);
```

Comment:
Well, may be add this assertion ("dbg" version) into `.elemSize()` methods directly?